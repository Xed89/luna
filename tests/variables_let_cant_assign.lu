type Program
  fun main()
    let n1 = 57
    n1 = 68