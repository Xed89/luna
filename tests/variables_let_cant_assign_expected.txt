Parsing completed
Syntax Tree:
TypeDeclaration 'Program'
  [0]: 
    Function 'main' (static)
      type:
        null
      statements:
        [0]: 
          DeclarationStatementSyntax
            let n1
              ExpressionLiteralSyntax '57'
        [1]: 
          VarOrCallChainMaybeAssignStatementSyntax
            VarOrCallChainSyntax
              [0]: 
                VariableOrCall 'n1'
              ExpressionLiteralSyntax '68'

Compile failed:
Variable n1 can't be modified because declared with 'let'

